exporter/datadogexporter/internal/clientutil/api.go

```diff
@@ -15,13 +15,70 @@
 package clientutil // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/clientutil"
 
 import (
+	"context"
 	"errors"
+	"os"
 
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV1"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
 	"go.uber.org/zap"
 	zorkian "gopkg.in/zorkian/go-datadog-api.v2"
 )
 
+// CreateAPIClient creates a new Datadog API client
+func CreateAPIClient(buildInfo component.BuildInfo, endpoint string, settings exporterhelper.TimeoutSettings, insecureSkipVerify bool) *datadog.APIClient {
+	configuration := datadog.NewConfiguration()
+	configuration.UserAgent = UserAgent(buildInfo)
+	configuration.HTTPClient = NewHTTPClient(settings, insecureSkipVerify)
+	configuration.Compress = true
+	if endpoint != "" {
+		configuration.Servers = datadog.ServerConfigurations{
+			{
+				URL:         "{site}",
+				Description: "No description provided",
+				Variables:   map[string]datadog.ServerVariable{"site": {DefaultValue: endpoint}},
+			},
+		}
+	}
+	return datadog.NewAPIClient(configuration)
+}
+
+// ValidateAPIKey checks if the API key (not the APP key) is valid
+func ValidateAPIKey(ctx context.Context, apiKey string, logger *zap.Logger, apiClient *datadog.APIClient) error {
+	logger.Info("Validating API key.")
+	authAPI := datadogV1.NewAuthenticationApi(apiClient)
+	resp, _, err := authAPI.Validate(GetRequestContext(ctx, apiKey))
+	if err == nil && *resp.Valid {
+		logger.Info("API key validation successful.")
+		return nil
+	}
+	if err != nil {
+		logger.Warn("Error while validating API key", zap.Error(err))
+		return nil
+	}
+	logger.Warn(ErrInvalidAPI.Error())
+	return ErrInvalidAPI
+}
+
+// GetRequestContext creates a new context with API key for DatadogV2 requests
+func GetRequestContext(ctx context.Context, apiKey string) context.Context {
+	if _, ok := os.LookupEnv("DD_API_KEY"); ok {
+		return datadog.NewDefaultContext(ctx)
+	}
```

Comment:
Why do we need this? If possible, we should ignore environment variables in the exporter code and leave that to the configuration resolver system (so here, in practical terms, never use `DD_API_Key` but `cfg.Api.Key` instead)